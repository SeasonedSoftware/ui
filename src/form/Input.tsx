import React from 'react'
import { TextField } from '@material-ui/core'
import startCase from 'lodash/startCase'

export default ({
  children,
  disabled,
  name,
  error,
  helper = ' ',
  label = startCase(name),
  placeholder = label,
  ...props
}: any) => {
  const helperText = error || helper
  return (
    <TextField
      {...props}
      name={name}
      id={name}
      label={label}
      placeholder={placeholder}
      fullWidth
      error={!!error}
      helperText={helperText}
      disabled={disabled}
      style={{
        marginBottom: '8px',
      }}
    />
  )
}
