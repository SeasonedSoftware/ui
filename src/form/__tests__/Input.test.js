import React from 'react'
import renderer from 'react-test-renderer'

import Input from '../Input'

it('renders correctly', () => {
  const props = {
    label: 'Foobar',
    name: 'foobar',
  }
  const tree = renderer.create(<Input {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
