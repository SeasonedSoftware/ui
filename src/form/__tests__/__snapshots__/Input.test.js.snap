// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  style={
    Object {
      "marginBottom": "8px",
    }
  }
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
    htmlFor="foobar"
    id="foobar-label"
  >
    Foobar
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-describedby="foobar-helper-text"
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      id="foobar"
      name="foobar"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Foobar"
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root"
    id="foobar-helper-text"
  >
     
  </p>
</div>
`;
